packages/shared/lib/services/proxy.service.ts

```diff
@@ -1,80 +1,100 @@
-            }
-        }
+    ): Promise<RouteResponse & Activities> {
+        const { success, error, response: proxyConfig, activityLogs } = this.configure(externalConfig, internalConfig);
+        if (!success || error || !proxyConfig) {
+            throw new Error(`Proxy configuration is missing: ${error}`);
+        }
+        return {
+            response: await this.sendToHttpMethod(proxyConfig, internalConfig),
+            activityLogs
+        };
+    }
 
+    public configure(
+        externalConfig: ApplicationConstructedProxyConfiguration | UserProvidedProxyConfiguration,
+        internalConfig: InternalProxyConfiguration
+    ): ServiceResponse<ApplicationConstructedProxyConfiguration> & Activities {
+        const activityLogs: ActivityLogMessage[] = [];
         let data = externalConfig.data;
         const { endpoint: passedEndpoint, providerConfigKey, connectionId, method, retries, headers, baseUrlOverride } = externalConfig;
-        const { environmentId: environment_id, accountId: optionalAccountId, isFlow, existingActivityLogId: activityLogId, isDryRun } = internalConfig;
-        const accountId = optionalAccountId ?? ((await environmentService.getAccountIdFromEnvironment(environment_id)) as number);
-        const logAction: LogAction = isFlow ? LogActionEnum.SYNC : LogActionEnum.PROXY;
-
-        let endpoint = passedEndpoint;
-        let connection: Connection | null = null;
-
-        // if this is a proxy call coming from a flow then the connection lookup
-        // is done before coming here. Otherwise we need to do it here.
-        if (!internalConfig.connection) {
-            const { success, error, response } = await connectionService.getConnectionCredentials(
-                accountId as number,
-                environment_id as number,
-                connectionId as string,
-                providerConfigKey as string,
-                activityLogId as number,
-                logAction,
-                false
-            );
+        const { connection, provider, existingActivityLogId: activityLogId } = internalConfig;
 
-            if (!success) {
-                if (throwErrors) {
-                    throw error;
-                } else {
-                    return { success: false, error, response: null };
-                }
+        if (!passedEndpoint && !baseUrlOverride) {
+            if (activityLogId) {
+                activityLogs.push({
+                    level: 'error',
+                    environment_id: connection.environment_id,
+                    activity_log_id: activityLogId,
+                    timestamp: Date.now(),
+                    content: 'Proxy: a API URL endpoint is missing.'
+                });
+            }
+            return { success: false, error: new NangoError('missing_endpoint'), response: null, activityLogs };
+        }
+        if (!connectionId) {
+            if (activityLogId) {
```

Comment:
A general feedback, it seems `ActivityLogMessage` do accepts undefined `activity_log_id` and there are two other checks later in the code at `createActivityLogMessageAndEnd()` and after `proxyService.configure`.
Maybe it's not necessary to have those `if` everywhere and would clean the code a bit.

Reply:
You are right, checking for activityLogId here is not necessary. I will fix